src/avro/src/schema.rs

```diff
@@ -1733,11 +1733,19 @@ impl<'a> Serialize for SchemaSerContext<'a> {
                         )?;
                         map.end()
                     }
-                    SchemaPiece::Enum { symbols, .. } => {
+                    SchemaPiece::Enum {
+                        symbols,
+                        default_idx,
+                        ..
+                    } => {
                         let mut map = serializer.serialize_map(None)?;
                         map.serialize_entry("type", "enum")?;
                         map.serialize_entry("name", &name)?;
                         map.serialize_entry("symbols", symbols)?;
+                        if let Some(default_idx) = *default_idx {
+                            assert!(default_idx < symbols.len());
```

Comment:
I see that we have a few places in `src/avro/src/schema.rs` where we're panicking instead of throwing an error, why do we do that?

Reply:
We should only do that in cases where we expect the condition to be impossible, i.e., it would only happen due to a logic bug in Materialize, not from any possible user input.

In this case, default_idx comes from the list of symbols so it can't possibly be out of bounds: https://github.com/MaterializeInc/materialize/blob/main/src/avro/src/schema.rs#L926